tests/State/HubAdapterTest.php

```diff
@@ -105,6 +105,23 @@ public function testWithScope(): void
         HubAdapter::getInstance()->withScope($callback);
     }
 
+    public function testWithScopeWithReturnValue(): void
+    {
+        $callback = static function () {
+            return true;
+        };
+
+        $hub = $this->createMock(HubInterface::class);
+        $hub->expects($this->once())
+            ->method('withScope')
+            ->with($callback)
+            ->willReturn(true);
```

Comment:
It would be wiser to use `willReturnArgument()` so that we won't have to bother changing the argument in a future refactoring if we will decide to change the expected value